### ISSUE TITLE
Nicks that are already taken

### ISSUE THREAD

**Issue by crookedneighbor** [haxiom/haxfred-irc#3]
I noticed that when Haxfred tried to rename himself EmperorZurg from his array of names, the console.log messages seemed to indicate that his renaming would be successful, but it wasn't, he renamed himself Haxfred2 instead. 

After some testing, I determined that the problem was that EmperorZurg was a reserved nick. Trouble is, if Haxfred hits a reserved name, he won't continue onto the next name, he gets stuck as Haxfred#. 

We can check to see if the name change was successful by checking the expected name with the actual name after Haxfred sends out the nick request, but is there an easy way to cycle through the available names again, starting after the offending name?

**Comment by lejeunerenard** [comment]
Besides the offending "EmperorZurg" nick, another part to this problem occurs when trying to start up a second haxfred in a different channel. He will name himself Haxfred2 instead of an alternative name. 

This is because Haxfred checks available names based on the channels current connected nicks, not the server's nick list. I'm not saying to get a list of all nicks on the server (if that is even possible), but instead we need to check if a name change was successful and move to the next nick if it fails.

To recap, the two problem scenarios are:
- Reserved Nicks
- Nicks taken on the server but arent in the current channel
